### PR TITLE
Add get_tracing_count feature

### PR DESCRIPTION
**Pull request by djdongjin**
This PR solved #37323.

### CODE REVIEW

**Review by jaingaurav — CHANGES_REQUESTED** [review]
Seems like there are 2 different fixes here. Could you please file separate PRs?

File: tensorflow/python/eager/def_function.py [hunk 1]
```diff
@@ -557,6 +557,13 @@ def _get_tracing_count(self):
     result += self._stateful_fn.tracing_count if self._stateful_fn else 0
     return result
 
+  def experimental_get_tracing_count(self):
+    """
+    Return the tracing count in the recent calls for debugging.
+    Feature request in issue: https://github.com/tensorflow/tensorflow/issues/37323
```
Comment (jaingaurav) [inline tensorflow/python/eager/def_function.py]: I don't think there is any need to reference the github issue in the comment.

**Comment by djdongjin** [comment]
> Seems like there are 2 different fixes here. Could you please file separate PRs?

No problem. Should I close this PR and file two other new PRs, or just move one-issue related commits to another new pr?

File: tensorflow/python/keras/optimizer_v2/adagrad.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def from_config(cls, config, custom_objects=None):
         An optimizer instance.
     """
     if 'initial_accumulator_value' not in config:
-      config['initial_accumulator_value'] = 0.
+      config['initial_accumulator_value'] = 0.1
```
Comment (jaingaurav) [inline tensorflow/python/keras/optimizer_v2/adagrad.py]: Were was the 0.1 default set for other types?

**Review by jaingaurav — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/optimizer_v2/adagrad.py (hunk 2 above)
Comment (djdongjin) [reply tensorflow/python/keras/optimizer_v2/adagrad.py]: yes, searched for the codebase and all other places use 0.1.

File: tensorflow/python/eager/def_function.py (hunk 1 above)
Comment (djdongjin) [reply tensorflow/python/eager/def_function.py]: fixed.

**Comment by jaingaurav** [comment]
@djdongjin: Code looks good. It might just look cleaner to setup some new PRs, but I'll leave that up to you.

**Comment by djdongjin** [comment]
> @djdongjin: Code looks good. It might just look cleaner to setup some new PRs, but I'll leave that up to you.

Thanks for review :). Given the size of this PR, I just revert a commit and move it to another pr #37820.

**Review by jaingaurav — DISMISSED** [review]

File: tensorflow/python/eager/def_function.py [hunk 3]
```diff
@@ -557,6 +557,12 @@ def _get_tracing_count(self):
     result += self._stateful_fn.tracing_count if self._stateful_fn else 0
     return result
 
+  def experimental_get_tracing_count(self):
```
Comment (kkimdev) [inline tensorflow/python/eager/def_function.py]: `_get_tracing_count(...)` is only used in this file.  Could you instead rename `_get_tracing_count` to `experimental_get_tracing_count` in this file? (with your function comment)

Comment (djdongjin) [reply tensorflow/python/eager/def_function.py]: fixed.

File: tensorflow/python/eager/def_function.py [hunk 4]
```diff
@@ -552,7 +552,10 @@ def _decorate(self, decorator):
     self._function_spec = function_lib.FunctionSpec.from_function_and_signature(
         self._python_function, self.input_signature)
 
-  def _get_tracing_count(self):
+  def experimental_get_tracing_count(self):
```
Comment (alextp) [inline tensorflow/python/eager/def_function.py]: If we're adding a new API this needs better documentation (including an example of how to use it) and a unit test.

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/eager/def_function.py (hunk 4 above)
Comment (djdongjin) [reply tensorflow/python/eager/def_function.py]: examples and unittests added.

Comment (alextp) [reply tensorflow/python/eager/def_function.py]: Can the example be made into a doctest as per https://www.tensorflow.org/community/contribute/docs_ref ?

Comment (alextp) [reply tensorflow/python/eager/def_function.py]: Also I don't know if users will understand what "tracing count" is. Maybe say "how many times the function has been traced"? Also mention again some conditions under which tf.function will do retraces.

Comment (djdongjin) [reply tensorflow/python/eager/def_function.py]: ok, will fix it shortly.

Comment (alextp) [reply tensorflow/python/eager/def_function.py]: We also need to be very clear here that people should not assume the retracing conditions are static; future versions of TF might need to retrace more or less often.

**Comment by gbaned** [comment]
@djdongjin Can you please resolve conflicts? Thanks!

**Pull request closed without merging**